docs/design/git-submodule-storage.md

```diff
@@ -59,22 +59,88 @@ These extensions are:
 - Colocated Git repos
 - Non-git backends
 
-## Possible approaches
+## Proposed design
 
-### Approach 1: Store Git submodules as full jj repos
+Git submodules will be stored as full jj repos. In the code, jj commands will
+only interact with the submodule's repo as an entire unit, e.g. it cannot query
+the submodule's commit backend directly. A well-abstracted submodule will extend
+well to non-git backends and non-git subrepos.
```

Comment:
> I suppose it will be possible to interact with the submodule's abstracted commit backend

Ah, I actually was thinking that we should try to avoid that if possible. From https://github.com/martinvonz/jj/pull/1482#discussion_r1173285661, I'm guessing you're thinking that e.g. to populate the working copy we would just read the submodule's commits off its backend directly, whereas I'm thinking of something more akin to "telling the submodule" to populate the working copy at whatever path we tell it to.

But at any rate, I don't think this should block progress. This will come down to what the code looks like and it's a pretty low stakes decision, so I'll just pick one approach and update this doc as we go along :)